programs/mmm/src/instructions/mip1/sol_mip1_fulfill_buy.rs

```diff
@@ -205,6 +214,44 @@ pub fn handler<'info>(
     )
     .map_err(|_| MMMErrorCode::NumericOverflow)?;
 
+    // check creator_accounts and verify the remaining accounts
+    let creator_accounts = if pool.using_shared_escrow() {
+        // check the remaining accounts at position 0 and 1
+        // 0 has to be the m2_program
+        // 1 has to be the shared_escrow_account pda of the m2_program
+        if ctx.remaining_accounts.len() < 2 {
+            return Err(MMMErrorCode::InvalidRemainingAccounts.into());
+        }
+        if *ctx.remaining_accounts[0].key != M2_PROGRAM {
+            return Err(MMMErrorCode::InvalidRemainingAccounts.into());
+        }
+        let shared_escrow_account = &ctx.remaining_accounts[1];
+        let pda_program =
+            check_buyside_sol_escrow_account(&shared_escrow_account.key(), &pool_key, &pool.owner)?;
+        if pda_program != M2_PROGRAM {
+            return Err(MMMErrorCode::InvalidRemainingAccounts.into());
+        }
+
+        // TODO
+        // Change to the correct amount to be withdrawn from shared escrow
+        // into the buyside_sol_escrow_account
+        let amount: u64 = (total_price as i64 + maker_fee) as u64;
+        withdraw_m2(
+            pool,
+            ctx.bumps.pool,
+            buyside_sol_escrow_account,
+            shared_escrow_account,
+            system_program,
+            &ctx.remaining_accounts[0],
```

Comment:
@samuelvanderwaal gave me a good suggestion when I did the m2 SPL changes, we can use a macro to make sure that this throws a anchor error instead of a system error when the indexing fails. Check out the `index_ra!` macro in m2